src/main/java/com/google/acai/TestScope.java

```diff
@@ -16,31 +16,33 @@
 
 package com.google.acai;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.inject.AbstractModule;
 import com.google.inject.Key;
 import com.google.inject.OutOfScopeException;
 import com.google.inject.Provider;
 import com.google.inject.Scope;
 
-import java.util.HashMap;
 import java.util.Map;
-
-import static com.google.common.base.Preconditions.checkState;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Scope for bindings annotated with {@link TestScoped}.
  */
 class TestScope implements Scope {
-  private final ThreadLocal<Map<Key<?>, Object>> values = new ThreadLocal<>();
+  private AtomicReference<ConcurrentMap<Key<?>, Object>> values = new AtomicReference<>();
```

Comment:
The situation is that I have an in-process server that handles requests on separate threads and uses an in-process database.  I issue several requests / test and want separate server state / test.

I didn't think about running tests in parallel.  This changes things.  For my case, I suppose I would want to either:
1) create a separate server/db per test
2) use the same server, just use separate dbs / test
3) run separate JVMs per test

I'll have to revise my solution (and maybe drop it).  InheritableThreadLocal<ConcurrentMap> might turn out to be useful, if each test created its own threads, or it might turn out to be just more confusing.  I'll go ahead and drop the request for now.